src/main/java/network/brightspots/rcv/GuiConfigController.java

```diff
@@ -747,6 +749,41 @@ public void buttonDeleteCvrFileClicked() {
         .removeAll(tableViewCvrFiles.getSelectionModel().getSelectedItems());
   }
 
+  /**
+   * Action when "Auto-Load Candidates" clicked.
+   */
+  public void buttonAutoLoadCandidatesClicked() {
+    Set<String> unloadedNames = new HashSet<>();
+    for (CvrSource source : tableViewCvrFiles.getItems()) {
+      ContestConfig config =
+            ContestConfig.loadContestConfig(createRawContestConfig(), FileUtils.getUserDirectory());
+      Provider provider = ContestConfig.getProvider(source);
+      try {
+        BaseCvrReader reader = provider.constructReader(config, source);
+        unloadedNames.addAll(reader.gatherUnknownCandidateNames());
+      } catch (ContestConfig.UnrecognizedProviderException e) {
+        Logger.severe("Unrecognized reader: %s", e.getMessage());
+      } catch (CastVoteRecord.CvrParseException | IOException e) {
+        Logger.severe("Failed to read file %s", source.getFilePath());
+      }
+    }
+
+    int successCount = 0;
+    for (String name : unloadedNames) {
+      Candidate candidate = new Candidate(name, "", false);
+      Set<ValidationError> validationErrors =
+              ContestConfig.performBasicCandidateValidation(candidate);
+      if (validationErrors.isEmpty()) {
+        tableViewCandidates.getItems().add(candidate);
+        successCount++;
+      } else {
+        Logger.warning("Candidate %s failed to load.", name);
```

Comment:
So this would only happen if the name is the empty string? Would the parser not have already choked on that?

Reply:
Not sure I follow -- this is checking if `validationErrors` is empty, not the `name`. I don't know if there are any checks preventing an empty-string candidate name -- perhaps it's an acceptable, unique name?

Reply:
See `performBasicCandidateValidation`

Reply:
Ah, I see -- you're right, that's the only thing checked in `performBasicCandidateValidation`. The parsers do not check for it, best I can see.

It still seems good to check for this here in case `performBasicCandidateValidation` adds additional checks in the future, just so the auto- and manual- candidate loader use the same validation step.